api-2.2/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ConditionVerificationStatusTranslatorImpl_2_2.java

```diff
@@ -26,28 +26,29 @@ public CodeableConcept toFhirResource(ConditionVerificationStatus verificationSt
 		if (verificationStatus == null) {
 			return null;
 		}
-		return this.setVerificationStatus(verificationStatus.toString());
+		return new CodeableConcept().addCoding(addVerificationStatusCoding(verificationStatus.toString()));
 	}
 	
 	@Override
 	public ConditionVerificationStatus toOpenmrsType(CodeableConcept codeableConcept) {
-		return codeableConcept.getCoding().stream().filter(coding -> coding.getSystem().equals(FhirConstants.OPENMRS_URI))
-		        .map(this::getVerificationStatus).findFirst().get();
-	}
-	
-	private CodeableConcept setVerificationStatus(String text) {
-		CodeableConcept codeableConcept = new CodeableConcept();
-		codeableConcept.addCoding().setCode(text).setDisplay(text).setSystem(FhirConstants.OPENMRS_URI);
-		return codeableConcept;
-	}
-	
-	private ConditionVerificationStatus getVerificationStatus(Coding coding) {
-		switch (coding.getCode().trim().toLowerCase()) {
-			case "confirmed":
+		if (codeableConcept.getCoding().isEmpty()) {
+			return null;
+		}
+		switch (codeableConcept.getCoding().get(0).getCode()) {
```

Comment:
We should probably normalize the case here, i.e., `codeableConcept.getCoding().get(0).getCode().toUpperCase()`

Reply:
Absolutely!✅